course_catalog/etl/video.py

```diff
@@ -0,0 +1,66 @@
+"""video catalog ETL"""
+
+from django.conf import settings
+from googleapiclient.discovery import build
+from django.conf import settings
+
+class YouTubeApi:
+    """
+    Class interface to YouTube API calls
+    """
+
+    client = None
+
+    def __init__(self):
+        api_service_name = "youtube"
+        api_version = "v3"
+        developer_key = settings.YT_DEVELOPER_KEY
+
+        self.client = build(api_service_name, api_version, developerKey=developer_key)
+
+    def get_videos_for_channel_in_batches(self, channel_id, offered_by):
+        page_token = ""
+
+        while True:
+            request = self.client.search().list(
+                part="snippet",
+                channelId=channel_id,
+                maxResults=50,
+                pageToken=page_token
+            )
+
+            response = request.execute()
+            response["offered_by"] = offered_by
+
+         
+            for video in response['items']:
+                try:
+                    video_id = video['id']['videoId']
+                    video['url'] = 'https://www.youtube.com/watch?v=' + video_id
+                    video['snippet']['full_description'] = self.get_full_description(video_id)
+                except KeyError:
+                    print(video)
+
+            yield response
+
+            if 'nextPageToken' in response.keys():
```

Comment:
You can drop the `.keys()` call since that's a little more idiomatic in python
```suggestion
            if 'nextPageToken' in response:
```